util/Migrator/DbScripts/2023-07-11_00_CollectionManagePermission.sql

```diff
@@ -2,78 +2,82 @@
  * Add Manage permission to collections
  */
 
--- Drop procedures that use the SelectionReadOnlyArray type
-IF OBJECT_ID('[dbo].[Group_CreateWithCollections]') IS NOT NULL
+ -- To allow the migration to be re-run, drop any of the V2 procedures
+
+IF OBJECT_ID('[dbo].[CollectionUser_UpdateUsers_V2]') IS NOT NULL
 BEGIN
-    DROP PROCEDURE [dbo].[Group_CreateWithCollections]
+    DROP PROCEDURE [dbo].[CollectionUser_UpdateUsers_V2]
 END
 GO
 
-IF OBJECT_ID('[dbo].[CollectionUser_UpdateUsers]') IS NOT NULL
+IF OBJECT_ID('[dbo].[Group_UpdateWithCollections_V2]') IS NOT NULL
 BEGIN
-    DROP PROCEDURE [dbo].[CollectionUser_UpdateUsers]
+    DROP PROCEDURE [dbo].[Group_UpdateWithCollections_V2]
 END
 GO
 
-IF OBJECT_ID('[dbo].[Group_UpdateWithCollections]') IS NOT NULL
+IF OBJECT_ID('[dbo].[Collection_UpdateWithGroupsAndUsers_V2]') IS NOT NULL
 BEGIN
-    DROP PROCEDURE [dbo].[Group_UpdateWithCollections]
+    DROP PROCEDURE [dbo].[Collection_UpdateWithGroupsAndUsers_V2]
 END
 GO
 
-IF OBJECT_ID('[dbo].[Collection_UpdateWithGroupsAndUsers]') IS NOT NULL
+IF OBJECT_ID('[dbo].[OrganizationUser_UpdateWithCollections_V2]') IS NOT NULL
 BEGIN
-    DROP PROCEDURE [dbo].[Collection_UpdateWithGroupsAndUsers]
+    DROP PROCEDURE [dbo].[OrganizationUser_UpdateWithCollections_V2]
 END
 GO
 
-IF OBJECT_ID('[dbo].[OrganizationUser_CreateWithCollections]') IS NOT NULL
+IF OBJECT_ID('[dbo].[Group_CreateWithCollections_V2]') IS NOT NULL
 BEGIN
-    DROP PROCEDURE [dbo].[OrganizationUser_CreateWithCollections]
+    DROP PROCEDURE [dbo].[Group_CreateWithCollections_V2]
 END
 GO
 
-IF OBJECT_ID('[dbo].[OrganizationUser_UpdateWithCollections]') IS NOT NULL
+IF OBJECT_ID('[dbo].[OrganizationUser_CreateWithCollections_V2]') IS NOT NULL
 BEGIN
-    DROP PROCEDURE [dbo].[OrganizationUser_UpdateWithCollections]
+    DROP PROCEDURE [dbo].[OrganizationUser_CreateWithCollections_V2]
 END
 GO
 
-IF OBJECT_ID('[dbo].[Collection_CreateWithGroupsAndUsers]') IS NOT NULL
+IF OBJECT_ID('[dbo].[Collection_CreateWithGroupsAndUsers_V2]') IS NOT NULL
 BEGIN
-    DROP PROCEDURE [dbo].[Collection_CreateWithGroupsAndUsers]
+    DROP PROCEDURE [dbo].[Collection_CreateWithGroupsAndUsers_V2]
 END
 GO
 
-IF TYPE_ID('[dbo].[SelectionReadOnlyArray]') IS NOT NULL
+-- Create a new SelectionReadOnlyArray_V2 type with a new [Manage] column
+IF TYPE_ID('[dbo].[SelectionReadOnlyArray_V2]') IS NOT NULL
 BEGIN
-    DROP TYPE [dbo].[SelectionReadOnlyArray]
+    DROP TYPE [dbo].[SelectionReadOnlyArray_V2]
 END
 GO
 
-CREATE TYPE [dbo].[SelectionReadOnlyArray] AS TABLE (
+CREATE TYPE [dbo].[SelectionReadOnlyArray_V2] AS TABLE (
```

Comment:
I may be misunderstanding your question, but there is a statement above that checks if the type exists and deletes it if it does.

And yes, it has to come after all the sprocs that reference the new type are removed; otherwise, we're prevented from recreating the type.

Reply:
d'oh, I missed that 😁